src/gui/src/mainWindow.cpp

```diff
@@ -513,36 +512,25 @@ QMenu* MainWindow::findMenu(QStringList& path, QMenu* parent)
   QString compare_name = cleanupText(top_name);
   path.pop_front();
 
+  QList<QAction*> actions;
   if (parent == nullptr) {
-    // loop through menu bar
-    QMenu* menu = nullptr;
-    auto* menubar = menuBar();
-    for (auto* action : menubar->actions()) {
-      if (cleanupText(action->text()) == compare_name) {
-        menu = action->menu();
-      }
-    }
-
-    if (menu == nullptr) {
-      menu = menubar->addMenu(top_name);
-    }
-
-    return findMenu(path, menu);
+    actions = menuBar()->actions();
   } else {
-    // loop through parent
-    QMenu* menu = nullptr;
-    for (auto* action : parent->actions()) {
-      if (cleanupText(action->text()) == compare_name) {
-        menu = action->menu();
-      }
-    }
+    actions = parent->actions();
+  }
 
-    if (menu == nullptr) {
-      menu = parent->addMenu(top_name);
+  QMenu* menu = nullptr;
+  for (auto* action : actions) {
+    if (cleanupText(action->text()) == compare_name) {
+      menu = action->menu();
     }
+  }
 
-    return findMenu(path, menu);
+  if (menu == nullptr) {
+    menu = parent->addMenu(top_name);
```

Comment:
if parent is null then you used to have
menu = menubar->addMenu(top_name);

is that handled here somewhere?

Reply:
it was lost, but is back now. oversight on my part.